src/schema-builder/RdbmsSchemaBuilder.ts

```diff
@@ -243,10 +243,10 @@ export class RdbmsSchemaBuilder implements SchemaBuilder {
                         if (indexMetadata.isUnique !== tableIndex.isUnique)
                             return true;
 
-                        if (indexMetadata.isSpatial !== tableIndex.isSpatial)
+                        if (indexMetadata.isSpatial !== !!tableIndex.isSpatial)
```

Comment:
since `isSpatial` is a boolean, we shouldn't except here something else other than boolean. Couldn't we fix it on the side where non-boolean is written into those properties?

Reply:
Sorry for long delay.
Value is newer assigned(undefined), but we can use boolean type here(without undefined) and assign false if developer didn't specify that option when defining table schema(changes in TableIndex.ts).